lib/puppet/type.rb

```diff
@@ -1537,12 +1537,18 @@ def to_edges
             :event => self.class.events,
             :callback => method
           }
-          self.debug { "subscribes to #{related_resource.ref}" }
+          ## Corrected syntax of debug statement to reflect the way this was called.
+          # i.e. before, after, subscribe, notify
+          #self.debug { "subscribes to #{related_resource.ref}" }
+          self.debug { "#{self.class.name} to #{related_resource.ref}" }
```

Comment:
It might be worth doing something like this to get a bit better grammar in the final output (and to make sure the existing `subscribes to` doesn't change for people):

```ruby
self.debug do
  relation = match self.class.name
  when "subscribe"
    "subscribes"
  when "notify"
   "notifies"
  else
    self.class.name
  end

  "#{relation} to #{related_resource.ref}"
end
```

Reply:
I agree that does make it more readable.
I have made the changes.